.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Static analysis
+run-name: GitHub Actions for clang-tidy
+
+on: pull_request
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: |
+          git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+          git fetch --no-tags --no-recurse-submodules upstream "${{ github.event.pull_request.base.ref }}"
+          git submodule update --init --recursive
+
+      - name: Install clang-tidy
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang clang-tidy libldap2-dev curl libcurl4-openssl-dev bison libudev-dev libkrb5-dev
+
+      - name: Prepare compile_commands.json
+        run: |
+          cmake -B build -DCMAKE_INSTALL_PREFIX=../install -DWITH_DEBUG=1 -DDOWNLOAD_BOOST=1 -DWITH_BOOST=my_boost \
```

Comment:
* Put build directory out of source tree, say to `../build`.
* Do not use `-DWITH_DEBUG` directly, use `-DCMAKE_BUILD_TYPE=Debug|RelWithDebInfo`.
  BTW, it also makes sense to check both `Debug` and `RelWithDebInfo` configurations with clang-tidy.
* Put boost directory out of source, say `../boost`. It also makes sense to cache the boost tarball archive using `actions/cache@v3` action. 
* For Boolean CMake Options please use ON|OFF instead of ''1|0'
* For this particular scenario, I think you should use `-DWITH_SYSTEM_LIBS=ON` for every library possible. Having something `bundled` just extends build command JSON file with additional items that won't be used.
* Something is wrong with `WITH_CURL` option, it points to a binary - just remove it.

Reply:
- Out of source build dir - Done.
- Use a cache for boost - Done.
- Update boolean options to ON/OFF - Done.
- Use DWITH_SYSTEM_LIBS=ON - Done.
- Remove curl option - Done.

Regarding checking for both **Debug** and **RelWithDebInfo** builds, wouldn't checking only for **Debug** build cover all the code since **-DCMAKE_BUILD_TYPE=Debug** is equivalent to a superset of both **RelWithDebInfo** and **Release** options from a compile time static analysis is concerned?